Pull Req: Next fix105 fixedsigningkeylocation

ronalag opened the pull request:
Fixed issue blackberry/BB10-Webworks-Packager#105 by changing the path of the default signing key location to include the drive of the home directory

ronalag committed:
Fixed issue #105 [Package thinks that the signing keys are on teh 2nd drive if you have a machine with 2 drives]

ronalag committed:
Corrected indentation

nukulb commented:
no unit test for this bug fix.

ronalag committed:
Added unit tests for issue #105

nukulb commented:
@EricLeiLi code review?

EricLeiLi commented:
r+

ronalag committed:
Fixed Linting errors

ishneur commented:
SV&V verification passed.

nukulb commented:
@jkeshavarzi - you want to quickly look at this?

jamesjhedley commented on test/unit/lib/signing-helper.js:
```diff
@@ -27,6 +27,42 @@ describe("signing-helper", function () {
+
+        spyOn(os, "type").andReturn("windows");
+        spyOn(path, "existsSync").andCallFake(function (p) {
+            return p.indexOf("C:") !== -1;
```
The way this is coded, only the first signature key location will be tested (local settings).
You need to test the AppData location as well and ensure it also gets the home drive prepended.

jamesjhedley replied:
The above is still an issue.

jamesjhedley commented on test/unit/lib/signing-helper.js:
```diff
@@ -27,6 +27,42 @@ describe("signing-helper", function () {
     });
     
+    it("returns a path on windows that specifies the drive", function () {
+        process.env.HOMEPATH = "Users\\user";
```
This could potentially cause problems for our other tests. If they use process.env.HOMEPATH or process.env.HOMEDRIVE they will get these invalid values.

jamesjhedley replied:
Since we can't mock objects the only thing i can think of is to store the old values for process.env.HOMEPATH and process.env.HOMEDRIVE and re-set them when your test is over. Even this i don't really like... because if the test fails midway, the objects won't be reset.

ronalag committed:
Made the following changes:
1. Changed the layout of unit tests for signing helper.
2. Changed unit tests for signing helper to preserve properties that are mocked.

ronalag commented:
@tohman, @EricLeiLi, @jkeshavarzi  - can you code review my latest changes for blackberry/BB10-Webworks-Packager#105

ronalag committed:
Made the following changes:
1. Reworded the test names to make them more readable.
2. Grouped common spies in before each statements.

ronalag commented:
hey @jkeshavarzi, updated the test cases. Code Reveiw ?

jamesjhedley commented on test/unit/lib/signing-helper.js:
```diff
@@ -2,55 +2,139 @@ var testData = require('./test-data'),
+            expect(result).toContain("C:");
+        });
+
+        it("can find keys on C drive", function () {
```
This test does the exact same thing as the test above. Accidental copy?

ronalag committed:
Made the following changes:
1. Prepended test home paths with file separator.
2. Added comments to explain property preservation in unit tests.
3. Corrected test case to check for presence of home path.

ronalag commented:
hey @jkeshavarzi, i made the following corrections:

-added comments to the before each and after each functions in the windows suite
-added the "\" prefix to the user home
-corrected the accidental copy above

jamesjhedley commented:
Code review complete.
@ishneur please make sure this is tested on mac as well. We want to make sure we didn't break anything.

jamesjhedley commented:
Actually my mistake. The code that was changed is only in the windows logic. This shouldnt affect mac at all.
As long as we tested on XP and Win7, we should be good to merge.

ishneur commented:
I only verified it on a Windows 7 machine with 2 drives (the one on which the bug was found). My XP machine has 1 drive. Do we have an XP and a Mac machine with 2 drives?

jamesjhedley commented:
@erhorim @nukulb Do we have an XP machine with two drives that Igor can use to verify this on XP?

ishneur commented:
I used Henry's XP machine which has 2 drives and it works.
